### ISSUE TITLE
float format

### ISSUE THREAD

**Issue by laurayuwen** [kframework/k-legacy#1451]
Currently the builtin Float2String method prints float in e+ format:  instead of 0.0, I got 0e+00. 
This made some of my tests fail (not the same format as Java prints). Shall we get the traditional one back?
e.g.  instead of
      return value.toString();
we can have 
       return String.valueOf(value.doubleValue());
in method  printKFloat.

**Comment by andreistefanescu** [comment]
Alternatively, we can have 2 Float2String functions, one for each format.

**Comment by dwightguth** [comment]
So, I chose the way I did because that method represents floats /exactly/,
ie, it will never claim that a float is anything other than the precise
binary value it corresponds to. This is a very useful property in an
arbitrary-precision float library because it means that the conversion to
string and back causes no loss of precision. Meanwhile, even if we could
make K represent the exact syntax of floats used by the JVM, we cannot make
it do so for all languages. Rather than chase the impossible, I suggest we
create a 2-argument Float2String function which accepts an MPFR format
string. This has the advantage of being trivial to implement and also
provide a very broad ability to customize for individual languages.
